go-micro.go

```diff
@@ -0,0 +1,47 @@
+/*
+Go micro provides a pluggable library to build microservices.
+
+	import (
+		micro "github.com/micro/go-micro"
+	)
+
+	service := micro.New(
+		micro.Registry(r),
+		micro.Broker(b),
+	)
+
+	h := service.Server().NewHandler(&Greeter{})
+	service.Server().Handle(h)
+	service.Run()
+
+
+	req := service.Client.NewRequest(service, method, request)
+	rsp := response{}
+	err := service.Client.Call(req, rsp)
+
+*/
+
+package gomicro
+
+import (
+	"github.com/micro/go-micro/client"
+	"github.com/micro/go-micro/server"
+)
+
+type Service interface {
+	Client() client.Client
+	Server() server.Server
+	Run() error
+	String() string
+}
+
+type Option func(*Options)
+
+var (
+	HeaderPrefix   = "X-Micro-"
+	DefaultService = newService()
```

Comment:
Depends on how you actually initialise. Defaults in go-micro don't normally result in go routines being spun up or anything magic. Its just a bit like DefaultServer and DefaultClient in net/http. It can be replaced, it can be left alone. It should not interfere but point noted. Want it to be elegant as opposed to cruft.